eglib/src/gmisc-unity.c

```diff
@@ -0,0 +1,19 @@
+#ifdef PLATFORM_UNITY
+
+#include <stdlib.h>
+#include <glib.h>
+
+#include "Path-c-api.h"
+
+static const char *tmp_dir;
+
+const gchar *
+g_get_tmp_dir(void)
+{
+    if (tmp_dir == NULL)
+        tmp_dir = UnityPalGetTempPath();
```

Comment:
We're actually giving the libil2cpp OS layer Mono's allocator, so the the memory is allocated in the same way the Mono code we are replacing allocates it.